src/main/java/com/google/sps/servlets/Utils.java

```diff
@@ -86,6 +90,36 @@ static Calendar loadCalendarClient() throws IOException {
     return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential).build();
   }
 
+  // Query datastore with the current users email to retrieve user info entity
+  static Entity getCurrentUserEntity() {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    return results.asSingleEntity();
+  }
+
+  // Retrieve the family id from the user info and fetch their family entity from datastore
+  static Entity getCurrentFamilyEntity(Entity userInfoEntity) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    long familyID = (long) userInfoEntity.getProperty("familyID");
```

Comment:
what is the property actually returned as?

A `Long`? If so, no need to do the explicit conversion

Reply:
This just returns as a java.lang.Object so it is necessary to cast it into the correct type